### PR TITLE
manage lambdas

### LINKED ISSUE #1832: Indexed style in tables don't apply when defined as a lambda function (Markdown format)
### What went wrong? 🤔

See the code below.

Replacing the element definition by the one shown on the line below generates a table that has no style applied to the cells.

### Expected Behavior

The element definition should behave identically.

### Steps to Reproduce Issue
```
from random import randrange
from taipy.gui import Gui, Markdown

x_range = range(0, 11)
data = {"x": [randrange(10) for _ in x_range]}

def x_style(_, value):
    return f"col{value}"

page = Markdown(
    "<|{data}|table|style[x]=x_style|show_all|>",
    # "<|{data}|table|style[x]={lambda _, v: f'col{v}'}|show_all|>",
    style={
        ".col0": {"background-color": "##b0b0b0"},
        ".col1": {"background-color": "#a5a4b0"},
        ".col2": {"background-color": "#9994b3"},
        ".col3": {"background-color": "#8880b0"},
        ".col4": {"background-color": "#7a71b0", "color": "white"},
        ".col5": {"background-color": "#6b5fad", "color": "white"},
        ".col6": {"background-color": "#5e50ad", "color": "white"},
        ".col7": {"background-color": "#5141ab", "color": "white"},
        ".col8": {"background-color": "#412eab", "color": "white"},
        ".col9": {"background-color": "#2c16a8", "color": "white"},
        ".col10": {"background-color": "#1b02a8", "color": "white"},
    },
)

Gui(page=page).run()
```

### Screenshots

image

### Version of Taipy

develop

### Acceptance Criteria

- [ ] Ensure new code is unit tested, and check code coverage is at least 90%.
- [ ] Create related issue in taipy-doc for documentation and Release Notes.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
resolves #1832

```
from random import randrange

from taipy.gui import Gui, Markdown

x_range = range(0, 11)
data = {"x": [randrange(10) for _ in x_range]}

page = Markdown(
     "<|{data}|table|style[x]={lambda _, v: f'col{v}'}|show_all|>",
    style={
        ".col0": {"background-color": "##b0b0b0"},
        ".col1": {"background-color": "#a5a4b0"},
        ".col2": {"background-color": "#9994b3"},
        ".col3": {"background-color": "#8880b0"},
        ".col4": {"background-color": "#7a71b0", "color": "white"},
        ".col5": {"background-color": "#6b5fad", "color": "white"},
        ".col6": {"background-color": "#5e50ad", "color": "white"},
        ".col7": {"background-color": "#5141ab", "color": "white"},
        ".col8": {"background-color": "#412eab", "color": "white"},
        ".col9": {"background-color": "#2c16a8", "color": "white"},
        ".col10": {"background-color": "#1b02a8", "color": "white"},
    },
)

Gui(page=page).run(title="1832 indexed styles lambda")

```

```
import pandas as pd

from taipy.gui import Gui

data = pd.DataFrame()

# Create 1O columns with 100 rows of bools
for i in range(10):
    data[f"col_{i}"] = list(range(100))

page = """
<|{data}|table|format_fn[col_2]={lambda s, v, i, r, c: f"value: {v} for {i} in {c}"}|not editable|>
"""
if __name__ == "__main__":
    Gui(page).run(title="564 table format function")

```

### CODE REVIEW

File: taipy/gui/_renderers/builder.py [hunk 1]
```diff
@@ -153,18 +153,22 @@ def _get_variable_hash_names(
...
+                if isinstance(v, str):
+                    looks_like_a_lambda = v.startswith("{lambda ") and v.endswith("}")
                     # need to unescape the double quotes that were escaped during preprocessing
                     (val, hash_name) = _Builder.__parse_attribute_value(gui, v.replace('\\"', '"'))
+                else:
+                    looks_like_a_lambda = False
+                    val = v
+                if callable(val):
+                    if val.__name__ == "<lambda>":
+                        if looks_like_a_lambda or not hash_name:
```
Comment (FabienLelaquais) [inline taipy/gui/_renderers/builder.py:170]: What would be the 'else' part?

File: taipy/gui/utils/_evaluator.py [hunk 2]
```diff
@@ -136,14 +140,15 @@ def __save_expression(
...
+        lambda_expr: t.Optional[bool] = False,
     ):
         if expr in self.__expr_to_hash:
             expr_hash = self.__expr_to_hash[expr]
             gui._bind_var_val(expr_hash, expr_evaluated)
             return expr_hash
         if expr_hash is None:
             expr_hash = _get_expr_var_name(expr)
-        else:
+        elif not lambda_expr:
```
Comment (FabienLelaquais) [inline taipy/gui/utils/_evaluator.py:152]: Not too sure about how and what happens if `lambda_expr` indeed,,,

**Review by FabienLelaquais — DISMISSED** [review]
Oh really?!?

**Review by namnguyen20999 — DISMISSED** [review]
Lgtm

File: taipy/gui/_renderers/builder.py (hunk 1 above)
Comment (FredLL-Avaiga) [reply taipy/gui/_renderers/builder.py:170]: if it is a lambda and it has already a hashname, we're fine
if it's not a callable and not a string) we don't care

File: taipy/gui/utils/_evaluator.py (hunk 2 above)
Comment (FredLL-Avaiga) [reply taipy/gui/utils/_evaluator.py:152]: if lambda expr, it has a hasname, we work with that

Comment (FabienLelaquais) [reply taipy/gui/utils/_evaluator.py:152]: Got it. May be a small comment stating so would help?

File: taipy/gui/_renderers/builder.py (hunk 1 above)
Comment (FabienLelaquais) [reply taipy/gui/_renderers/builder.py:170]: Got it. May be a small comment stating so would help?

**Review by namnguyen20999 — DISMISSED** [review]

File: taipy/gui/_renderers/builder.py [hunk 3]
```diff
@@ -153,18 +153,24 @@ def _get_variable_hash_names(
...
             if hash_name is None:
-                if callable(v):
-                    if v.__name__ == "<lambda>":
-                        hash_name = f"__lambda_{id(v)}"
-                        gui._bind_var_val(hash_name, v)
-                    else:
-                        hash_name = _get_expr_var_name(v.__name__)
-                elif isinstance(v, str):
+                if isinstance(v, str):
+                    looks_like_a_lambda = v.startswith("{lambda ") and v.endswith("}")
```
Comment (dinhlongviolin1) [inline taipy/gui/_renderers/builder.py:160]: I like this naming very much 😃

File: taipy/gui/gui.py [hunk 4]
```diff
@@ -1652,6 +1652,9 @@ def _is_in_brdcst_callback(self):
         except RuntimeError:
             return False
 
+    def _get_lambda_id(self, lambda_fn: LambdaType):
```
Comment (dinhlongviolin1) [inline taipy/gui/gui.py]: Should make this an importable utils

**Review by dinhlongviolin1 — DISMISSED** [review]
Thank you for waiting for my approval

File: taipy/gui/builder/_element.py [hunk 5]
```diff
@@ -121,10 +121,10 @@ def __parse_lambda_property(self, key: str, value: t.Any) -> t.Any:
                 return None
             args = [arg.arg for arg in lambda_fn.args.args]
             targets = [
-                compr.target.id  # type: ignore[attr-defined]
+                comprehension.target.id  # type: ignore[attr-defined]
```
Comment (FabienLelaquais) [inline taipy/gui/builder/_element.py:124]: 👍

**Review by FabienLelaquais — APPROVED** [review]
👌

**Review by namnguyen20999 — APPROVED** [review]

**Pull request merged**